pangeo_forge_recipes/storage.py

```diff
@@ -37,10 +37,13 @@ def _copy_btw_filesystems(input_opener, output_opener, BLOCK_SIZE=10_000_000):
     with input_opener as source:
         with output_opener as target:
             while True:
-                data = source.read(BLOCK_SIZE)
-                if not data:
-                    break
-                target.write(data)
+                try:
+                    data = source.read(BLOCK_SIZE)
+                    if not data:
+                        break
+                    target.write(data)
+                except ValueError:
+                    raise BlockSizeError()
```

Comment:
When raising an exception from within an `except` block you should use `raise from e` to make sure that the traceback is formatted propertly.
```suggestion
                except ValueError as e:
                    raise BlockSizeError() from e
```

https://stefan.sofa-rockers.org/2020/10/28/raise-from/

Reply:
Thanks Tom